docs/testplan/dash/Dash-ENI-Based-Forwarding-test-plan.md

```diff
@@ -0,0 +1,163 @@
+Until HaMgrd is available, we can only write configuration to the DASH_ENI_FORWARD_TABLE.
+DASH_ENI_FORWARD_TABLE schema: https://github.com/r12f/SONiC/blob/user/r12f/ha2/doc/smart-switch/high-availability/smart-switch-ha-detailed-design.md#2321-dash_eni_forward_table
+
+Common tests configuration:
+- Test will be based on the basic private link configuration.
+- Need apply the configuration for DPU_TABLE, VIP_TABLE, VXLAN_TUNNEL, VNET in config_db according to the dash PL config.
+
+Common tests cleanup:
+- Remove the basic private link configuration.
+- Remove the configuration for DPU_TABLE, VIP_TABLE, VXLAN_TUNNEL, VNET in config_db.
+
+We need apply the config for DPU_TABLE, VIP_TABLE, VXLAN_TUNNEL and VNET into NPU config_db.
+Example:
+```
+{
+    "DPU_TABLE": {
+        "1": {
+            "type": "local",
+            "state": "up",
+            "pa_ipv4": "10.0.0.75",
+            "npu_ipv4": "10.1.0.32"
+        },
+        "2": {
+              "type": "cluster",
+              "state": "up",
+              "pa_ipv4": "10.0.0.79",
+              "npu_ipv4": "10.1.0.32"
+        }
+    },
+    "VIP_TABLE": {
+       "10.2.0.1/32" : {}
+    },
+    "VXLAN_TUNNEL": {
+        "tunnel1": {
+            "src_ip": "<Loopback of NPU>"
+        }
+    },
+    "VNET": {
+        "Vnet1000": {
+            "vxlan_tunnel": "tunnel1",
+            "vni": "1000"
+        }
+    }
+}
+```
+
+In a full functional HA testbed, the hamgrd should generate the entry in DASH_ENI_FORWARD_TABLE based on the above configuration. And this will be covered in the HA test.
+In this test, we are not going to deploy and test the full HA fuctionality, so we still need to apply the DASH_ENI_FORWARD_TABLE via swssconfig to the APPL_DB.
+Example:
+```
+[
+    {
+        "DASH_ENI_FORWARD_TABLE:Vnet1000:F4:93:9F:EF:C4:7E":
+        {
+            "vdpu_ids": "1,2",
+            "primary_vdpu": "1",
+            "outbound_vni": "1000",
+            "outbound_eni_mac_lookup": ""
+        },
+        "OP": "SET"
+    }
+]
+```
+The outbound_vni is optional, it indicates the outbound VNI used by this ENI if it's different from the one in VNET. Each ENI can have its own VNI, such as ExpressRoute Gateway Bypass case. If it's not assigned, this is the vni in the VNET.
+This field will be paramterized in the test by a pytest fixture, for the testcase uses this fixture, the outbound_vni will be tested both explicitly and implicitly. By default, we don't configure this field.
+This fixture will be used only in test case #1(test_privatelink_basic_transform migrate to ENI based fowarding) to save test run time.
+
+The outbound_eni_mac_lookup is used to decide whether we lookup the src mac or dst mac for ENI. Currenly the feature using this is not available yet. The validation for this field is not in the scope of this test plan. It can be removed from the config.
+
+
+## Test
+### Test case # 1 – test_privatelink_basic_transform migrate to ENI based fowarding
```

Comment:
Hi @congh-nvidia , after a second thought, we don't have to create a fixture/plugin for reusing the test cases. We can simply call the existing test functions for sending and validating the traffic. In this way, the top level test cases don't have to change, and also easy for future changes, say from DPU to ENI.

Reply:
Hi @r12f , according to the design changed in https://github.com/sonic-net/SONiC/pull/2081 and discussion we had with MSFT team, I have update the test plan and narrowed the test scope to only test on the NPU. The DPU dataplane interface is mocked by a NPU front panel interface.
In this case, we don't need consider the other existing dash tests and the functionality of this feature can be fully  covered in this test.
Please review the test plan, the test is also ready, I will create the PR soon.